rest/rest.go

```diff
@@ -19,7 +19,7 @@ type EndpointAlias struct {
 // EndpointAction represents an action on an API endpoint.
 type EndpointAction struct {
 	Handler        func(state *state.State, r *http.Request) response.Response
-	AccessHandler  func(state *state.State, r *http.Request) response.Response
+	AccessHandler  func(state *state.State, r *http.Request) (response.Response, func(w http.ResponseWriter) error)
```

Comment:
I'm not certain about this callback approach. I think it might be simpler to make this a
```
AccessHandler func(state *state.State, r *http.Request) (bool, response.Response)
```

The request can be allowed to continue if the returned value is true. `response.Response` is always rendered if not nil.

Reply:
the callback is more for writing to the `http.ResponseWriter`, currently if the request moves forward, the `response.Response` returned by `Handler` would be rendered. Which means we can't write to the response from the `AccessHandler`

Reply:
Yep I understand that, but the `response.Response` is already essentially a callback. I'm proposing that we no longer use it to decide whether or not the request should continue. The access handler can return:
- `true, nil` allow the request to continue, don't write anything to the response writer.
- `true, response.ManualResponse(...)`, allow the request to continue, call `Render` on the response.
- `false, nil`, stop handling the request, return a 403 with a generic forbidden message.
- `false, response.ManualResponse(...)`, stop handling the request, call `Render` on the response.

Reply:
Got you, that feels way cleaner :+1:

Reply:
Yeah that seems fair I think